dragonfly/item/hoe.go

```diff
@@ -0,0 +1,76 @@
+package item
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/internal/item_internal"
+	"github.com/df-mc/dragonfly/dragonfly/item/tool"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+type Hoe struct {
+	Tier tool.Tier
+}
+
+func (h Hoe) UseOnBlock(pos world.BlockPos, face world.Face, clickPos mgl64.Vec3, w *world.World, user User, ctx *UseContext) bool {
+	if grass := w.Block(pos); grass == item_internal.Grass {
+		if face == world.FaceDown {
+			// Tilled land isn't created when the bottom face is clicked.
+			return false
+		}
+		if w.Block(pos.Add(world.BlockPos{0, 1})) != item_internal.Air {
+			// Tilled land can only be created if air is above the grass block.
+			return false
+		}
+		w.SetBlock(pos, item_internal.FarmLand)
+		//TODO: Play the sound
+		ctx.DamageItem(1)
+		return true
+	}
+	return false
+}
+
+func (h Hoe) MaxCount() int {
+	return 1
+}
+
+func (h Hoe) AttackDamage() float64 {
+	return h.Tier.BaseAttackDamage
```

Comment:
The attack damage of hoes in Bedrock Edition is one higher than shovels, so return `h.Tier.BaseAttackDamage + 1` here.